mhlo: fix CMake build for tosa pipeline

ashay opened the pull request:
The include directories for building the MHLO and TOSA interop is
incorrect, causing the CMake build to fail.  This patch fixes the
include statements.

---

cc: @jpienaar This is likely to break the Bazel build, so I've marked this as a Draft PR, but let me know if there are any small fixes I can make so that the bazel build will pass.

tpopp commented:
Sorry, I just noticed this review. I'm going to assign to Jacques as a more appropriate person. As you predicted, this breaks the bazel build without further modification

ashay commented:
Thanks @tpopp!  At the cost of sounding naive, how do I trigger a bazel build in the mlir-hlo repo (in the hopes of fixing this myself)?  From what I see, the README includes instructions for CMake builds but not Bazel builds.

jpienaar commented:
I missed this one too (unfortunately the person most active on this section has been out for a month and will be out for one more).

@burmako  for visibility and there is the pending issue to externalize the cmake CI.

This repo is a direct mirror of what's in OpenXLA/XLA and TF, the bazel build instructions for the former probably applies here (but current recommendation there is to use Docker, which I'm not particularly found of as build step). @GMNGeoffrey for visibility too given current dependencies.

ashay committed:
build: use LINK_COMPONENTS for linking LLVMSupport

Without this patch, in-tree builds of LLVM+MHLO fail with the following
message:

```
MhloRngUtils specifies LINK_LIBS LLVMSupport, but LINK_LIBS cannot be used
for LLVM libraries.  Please use LINK_COMPONENTS instead.
```
